### ISSUE TITLE
In TAStudio recording mode, input can't be easily undone

### ISSUE THREAD

**Issue by RetroEdit** [TASVideos/BizHawk#2143]
When entering recording mode in TAStudio, the previous input will be overwritten. This is sometimes accidental, and it would be desirable to be able to undo the current session of recorded input, rather than undoing each input one-by-one.

Since I've already looked at the undo handling code, I might have a look at this; I already plan to do some more undo/redo revamping, since undo/redo insert inefficiency (#1988) wasn't the only slow thing.

**Comment by vadosnaprimer** [comment]
FCEUX handles this by combining consecutively recorded frames into a single Undo step.

**Comment by RetroEdit** [comment]
Yeah, that's exactly what I was thinking should happen. Architecturally speaking, it might require some modifications to BizHawk's undo/redo API, since I'm not sure it can handle actions occurring for multiple consecutive frames (doing stuff like deleting multiple frames, etc. is different because it can be handled as a batch action).

**Comment by SuuperW** [comment]
Fixed with #4382
